recipe/meta.yaml

```diff
@@ -20,22 +20,27 @@ outputs:
       run_exports:
         strong:
           - cudatoolkit {{ cuda_compiler_version }}|{{ cuda_compiler_version }}.*
-          - __glibc >=2.17  # [linux and cuda_compiler_version == "11.0"]
+          - __glibc >=2.17  # [linux and (cuda_compiler_version or "").startswith("11.")]
     requirements:
       host:
         # Needed to symlink libcuda into sysroot libs.
         - {{ compiler("c") }}
-        - sysroot_linux-64 2.17  # [linux and cuda_compiler_version == "11.0"]
-        - __glibc >=2.17         # [linux and cuda_compiler_version == "11.0"]
+        - sysroot_linux-64 2.17  # [linux and (cuda_compiler_version or "").startswith("11.")]
+        - __glibc >=2.17         # [linux and (cuda_compiler_version or "").startswith("11.")]
       run:
-        - sysroot_linux-64 2.17  # [linux and cuda_compiler_version == "11.0"]
-        - __glibc >=2.17         # [linux and cuda_compiler_version == "11.0"]
+        - sysroot_linux-64 2.17  # [linux and (cuda_compiler_version or "").startswith("11.")]
+        - __glibc >=2.17         # [linux and (cuda_compiler_version or "").startswith("11.")]
+      run_constrained:
+        - gcc_linux-64 <=10      # [linux and (cuda_compiler_version or "").startswith("11.1.")]
+        - gcc_linux-64 <=9       # [linux and (cuda_compiler_version or "").startswith("11.0.")]
+        - gcc_linux-64 <=8       # [linux and (cuda_compiler_version or "").startswith(("10.2.", "10.1."))]
+        - gcc_linux-64 <=7       # [linux and (cuda_compiler_version or "").startswith(("10.0.", "9.2."))]
```

Comment:
Using run constraints felt more appropriate than run dependencies to continue to allow people to bring their own compiler

Reply:
I don't think this will work as the version is already set in the recipe's build variants, which conflicts.

My guess is we will need to use [`zip_keys` with the `cxx_compiler_version` in `conda-forge-pinning`]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/53b26ee6df5eb51181b4ca74fdfb57940316be5c/recipe/conda_build_config.yaml#L131-L143 ) like what we have done already with `docker_image`. Though we can test this hypothesis out by adding a `conda_build_config.yaml` to the `recipe` here with the same change.

Reply:
I'm a bit lost as to how I would go about doing this. Would I create a new key with say `[9, 7, 7, 7, 9]` to match the `cuda_compiler_version` keys? How would I make `{{ compiler("c") }}` play nicely with that since it seems to already be set by the earlier zip key?

Reply:
I didn't manage to get that particular zip working: https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/1000